spec/amber/cli/commands/generator_spec.cr

```diff
@@ -6,32 +6,39 @@ include CLIHelper
 include CLIFixtures
 
 module Amber::CLI
-  begin
-    describe MainCommand::Generate do
-      context "scaffold" do
-        it "generates and compile generated app" do
-          ENV["AMBER_ENV"] = "test"
-          scaffold_app(TESTING_APP)
-          MainCommand.run ["generate", "scaffold", "Animal", "name:string"]
-          assert_app_compiled?(TEST_APP_NAME)
-          cleanup
-        end
+  describe MainCommand::Generate do
+    ENV["AMBER_ENV"] = "test"
+
+    context "when scaffolding an Amber app" do
+      scaffold_app(TESTING_APP)
+      MainCommand.run ["generate", "scaffold", "Animal", "name:string"]
+      `shards build`
+
+      it "generates and compile generated app" do
+        File.exists?("bin/#{TEST_APP_NAME}").should be_true
       end
 
-      context "controllers" do
-        it "generates controller with correct verbs and actions" do
-          ENV["AMBER_ENV"] = "test"
-          controller = "Animal"
-          options = %w(add:post list:get remove:delete)
-          scaffold_app(TESTING_APP)
+      cleanup
+    end
+
+    context "when generating controllers" do
+      controller = "Animal"
+      options = %w(add:post list:get remove:delete)
+      route_file = File.read("./config/routes.cr")
+      MainCommand.run ["generate", "controller", controller] | options
+
+      it "generates controller with correct verbs and actions" do
+        File.read("./src/controllers/#{controller.downcase}_controller.cr").should eq expected_controller
```

Comment:
This line and line 25 are coupling this code with wherever `expected_controller` is defined. I would say choose one place and have both `controller = "Animal"` and whatever is in the definition of `expected_controller`. I think the helper or the fixture is the better place.

Reply:
For an example of how I approached it, see here: https://github.com/amberframework/amber/pull/340/commits/3220d1312608f53ca486b8f5a668219870c19b57

And the helper here: https://github.com/marksiemers/amber/blob/3220d1312608f53ca486b8f5a668219870c19b57/spec/amber/cli/templates/migration_spec_helper.cr

All the decisions about names are in the helper - so they are easier to keep in sync.